File: pingdart-js/src/services/storage.service.js
class StorageService {
    constructor(http, config) {
        this.http = http;
    }

    /**
     * Get storage statistics.
     */
    async getStats() {
        try {
            const response = await this.http.post("/files/stats");
            return response.data;
        } catch (error) {
            throw new Error(`Failed to get storage stats: ${error.message}`);
        }
    }

    /**
     * List all buckets.
     */
    async listBuckets() {
        try {
            const response = await this.http.post("/files/get-buckets");
            return response.data;
        } catch (error) {
            throw new Error(`Failed to list buckets: ${error.message}`);
        }
    }

    /**
     * Create a new bucket.
     */
    async createBucket(name) {
        try {
            const response = await this.http.post("/files/create-bucket", { name });
            return response.data;
        } catch (error) {
            throw new Error(`Failed to create bucket: ${error.message}`);
        }
    }

    /**
     * Delete a bucket.
     */
    async deleteBucket(bucketId) {
        try {
            const response = await this.http.post("/files/delete-bucket", { id: bucketId });
            return response.data;
        } catch (error) {
            throw new Error(`Failed to delete bucket: ${error.message}`);
        }
    }

    /**
     * Upload a file.
     * @param {FormData} formData - The form data containing the file.
     */
    async uploadFile(formData) {
        try {
            const response = await this.http.post("/files/upload", formData, {
                headers: {
                    'Content-Type': 'multipart/form-data'
                }
            });
            return response.data;
        } catch (error) {
            throw new Error(`Failed to upload file: ${error.message}`);
        }
    }

    /**
     * Delete a file.
     */
    async deleteFile(filePath) {
        try {
            const response = await this.http.post("/files/deleteFile", { filePath });
            return response.data;
        } catch (error) {
            throw new Error(`Failed to delete file: ${error.message}`);
        }
    }
}

export default StorageService;

File: pingdart-python/pingdart_sdk/services/storage_service.py
class StorageService:
    def __init__(self, http_client):
        self.http = http_client

    def get_stats(self):
        """
        Get storage statistics.
        """
        try:
            response = self.http.post("/files/stats")
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to get storage stats: {str(e)}")

    def list_buckets(self):
        """
        List all buckets.
        """
        try:
            response = self.http.post("/files/get-buckets")
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to list buckets: {str(e)}")

    def create_bucket(self, name):
        """
        Create a new bucket.
        """
        try:
            response = self.http.post("/files/create-bucket", json={"name": name})
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to create bucket: {str(e)}")

    def delete_bucket(self, bucket_id):
        """
        Delete a bucket.
        """
        try:
            response = self.http.post("/files/delete-bucket", json={"id": bucket_id})
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to delete bucket: {str(e)}")

    def upload_file(self, file_path, bucket_name=None, destination=None):
        """
        Upload a file using multipart/form-data.
        """
        try:
            with open(file_path, 'rb') as f:
                files = {'file': f}
                data = {}
                if bucket_name:
                    data['bucket_name'] = bucket_name
                if destination:
                    data['destination'] = destination
                
                # We need to bypass our standard json request for multipart
                # self.http.headers will have x-api-key which is good
                response = self.http.request(
                    'POST', 
                    "/files/upload", 
                    files=files, 
                    data=data
                )
                response.raise_for_status()
                return response.json()
        except Exception as e:
            raise Exception(f"Failed to upload file: {str(e)}")

    def delete_file(self, file_path):
        """
        Delete a file.
        """
        try:
            response = self.http.post("/files/deleteFile", json={"filePath": file_path})
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to delete file: {str(e)}")
